backend/audit/test_schemas.py

```diff
@@ -1,20 +1,264 @@
 # Even though the schemas are not Django views or modules etc., we test them
 # here for CI/CD integration.
 import json
+import string
+
 from pathlib import Path
 from django.test import SimpleTestCase
-from jsonschema import exceptions, validate
+from jsonschema import exceptions, validate as jsonschema_validate, FormatChecker
+from random import choice, randrange
 
 # Simplest way to create a new copy of simple case rather than getting
 # references to things used by other tests:
 jsoncopy = lambda v: json.loads(json.dumps(v))
 
+
+# wrap the validate function to include a format checker
+def validate(instance, schema):
+    return jsonschema_validate(instance, schema, format_checker=FormatChecker())
+
+
 SCHEMA_DIR = Path(__file__).parent.parent / "schemas" / "sections"
 
 
-class SchemaValidityTest(SimpleTestCase):
+class GeneralInformationSchemaValidityTest(SimpleTestCase):
+    """
+    Test the basic validity of the GeneralInformation JSON schema.
+    """
+
+    GENERAL_INFO_SCHEMA = json.loads(
+        (SCHEMA_DIR / "GeneralInformation.schema.json").read_text(encoding="utf-8")
+    )
+
+    SIMPLE_CASE = {
+        "GeneralInformation": {
+            "auditee_fiscal_period_start": "2022-01-01",
+            "auditee_fiscal_period_end": "2022-12-31",
+            "ein": "123456789",
+            "ein_not_an_ssn_attestation": True,
+            "multiple_eins_covered": False,
+            "auditee_name": "John",
+            "auditee_address_line_1": "123 Fake St.",
+            "auditee_city": "FakeCity",
+            "auditee_state": "AL",
+            "auditee_zip": "12345",
+            "auditee_contact_name": "John",
+            "auditee_contact_title": "A Title",
+            "auditee_phone": "555-555-5555",
+            "auditee_email": "john@test.test",
+            "user_provided_organization_type": "state",
+            "met_spending_threshold": True,
+            "is_usa_based": True,
+            "auditor_firm_name": "Firm LLC",
+            "auditor_ein": "123456789",
+            "auditor_ein_not_an_ssn_attestation": True,
+            "auditor_country": "USA",
+            "auditor_address_line_1": "456 Fake St.",
+            "auditor_city": "AnotherFakeCity",
+            "auditor_state": "WY",
+            "auditor_zip": "56789",
+            "auditor_contact_name": "Jane",
+            "auditor_contact_title": "Another Title",
+            "auditor_phone": "999-999-9999",
+            "auditor_email": "jane@test.test",
+        }
+    }
+
+    def test_simple_pass(self):
+        """
+        The simple case should be valid
+        """
+        schema = self.GENERAL_INFO_SCHEMA
+
+        validate(self.SIMPLE_CASE, schema)
+
+    def test_invalid_auditee_fiscal_period_start(self):
```

Comment:
Undecided on whether these tests add any value- in one sense, they're testing `jsonschema's` internal date validation functionality (which isn't valuable). In another sense, they're testing that we are in fact marking this field as needing to be date-formatted (which is valuable).

Reply:
It's probably better to keep them as an additional safety net against accidentally not marking a field as a date.